codegen/service/service_data.go

```diff
@@ -738,6 +738,7 @@ func collectProjectedTypesR(projected, att *design.AttributeExpr, seen map[strin
 	collect := func(projected, att *design.AttributeExpr) []*ProjectedTypeData {
 		return collectProjectedTypesR(projected, att, seen, scope, viewspkg)
 	}
+	projected.ForcePointer = true
```

Comment:
We don't need to recurse here? ^

Reply:
This is already in a recursive logic which goes through all attributes to collect projected types.